compiling error - recipe for target failed

Dear Author,

I am having trouble compiling, when i issue 'make' I receive the errors ->

Making all in src
make[1]: Entering directory '/home/james/AlsaJsonGateway/src'
  CC       ajg_daemon-alsa-ajg.o
In file included from /usr/include/string.h:635:0,
                 from ../include/local-def-ajg.h:29,
                 from alsa-ajg.c:29:
In function ‘strncat’,
    inlined from ‘alsaFakeResponse.isra.0’ at alsa-ajg.c:60:12:
/usr/include/x86_64-linux-gnu/bits/string3.h:156:10: error: call to __builtin___strncat_chk might overflow destination buffer [-Werror]
   return __builtin___strncat_chk (__dest, __src, __len, __bos (__dest));
          ^
cc1: all warnings being treated as errors
Makefile:530: recipe for target 'ajg_daemon-alsa-ajg.o' failed
make[1]: *** [ajg_daemon-alsa-ajg.o] Error 1
make[1]: Leaving directory '/home/james/AlsaJsonGateway/src'
Makefile:399: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

Would you please help me to understand the error and fix it? I would love to be able to use my 18i20 with Linux. 

Best wishes,

fisch

--- a/src/alsa-ajg.c
+++ b/src/alsa-ajg.c
@@ -1,7 +1,7 @@
 /*
    alsajson-gw -- provide a REST/HTTP interface to ALSA-Mixer
 
-   Copyright (C) 2015, Fulup Ar Foll
+   Copyright (C) 2018, Fulup Ar Foll, Google LLC
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -40,33 +40,33 @@ STATIC json_object *alsaFakeResponse (AJG_session *session, AJG_request *request
     char filename[256];
 
     strncpy (filename, session->config->rootdir, sizeof(filename));
-    strncat (filename, "/fakemod/", sizeof(filename));
+    strncat (filename, "/fakemod/", sizeof(filename) - 1);
 
     switch (fakecmd) {
      case  CARD_GET_NAME:
-           strncat (filename, "CARD_GET_NAME", sizeof(filename));
+           strncat (filename, "CARD_GET_NAME", sizeof(filename) - 1);
            break;
      case  GATEWAY_PING:
-           strncat (filename, "GATEWAY_PING", sizeof(filename));
+           strncat (filename, "GATEWAY_PING", sizeof(filename) - 1);
            break;
      case  CARD_GET_ALL:
-           strncat (filename, "CARD_GET_ALL", sizeof(filename));
+           strncat (filename, "CARD_GET_ALL", sizeof(filename) - 1);
            break;
      case  CARD_GET_ONE:
-           strncat (filename, "CARD_GET_ONE", sizeof(filename));
+           strncat (filename, "CARD_GET_ONE", sizeof(filename) - 1);
            break;
      case  CTRL_GET_ALL:
-           strncat (filename, "CTRL_GET_ALL-", sizeof(filename));
-           strncat (filename, request->cardid, sizeof(filename));
+           strncat (filename, "CTRL_GET_ALL-", sizeof(filename) - 1);
+           strncat (filename, request->cardid, sizeof(filename) - 1);
            break;
      case  CTRL_GET_ONE:
-           strncat (filename, "CTRL_GET_ONE", sizeof(filename));
+           strncat (filename, "CTRL_GET_ONE", sizeof(filename) - 1);
            break;
      case  CTRL_SET_ONE:
-           strncat (filename, "CTRL_SET_ONE", sizeof(filename));
+           strncat (filename, "CTRL_SET_ONE", sizeof(filename) - 1);
            break;
      case  CTRL_SET_MANY:
-           strncat (filename, "CTRL_SET_MANY", sizeof(filename));
+           strncat (filename, "CTRL_SET_MANY", sizeof(filename) - 1);
            break;
 
      default:
@@ -74,7 +74,7 @@ STATIC json_object *alsaFakeResponse (AJG_session *session, AJG_request *request
     }
 
     // just upload json object and return without any further processing
-    strncat (filename, ".ajg", sizeof(filename));
+    strncat (filename, ".ajg", sizeof(filename) - 1);
     fakeResponse = json_object_from_file (filename);
 
     // if not fakemessage for this request build an error message

--- a/src/httpd-ajg.c
+++ b/src/httpd-ajg.c
@@ -1,7 +1,7 @@
 /*
    alsajson-gw -- provide a REST/HTTP interface to ALSA-Mixer
 
-   Copyright (C) 2015, Fulup Ar Foll
+   Copyright (C) 2018, Fulup Ar Foll, Google LLC
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -366,8 +366,8 @@ STATIC int requestFile (struct MHD_Connection *connection, AJG_session *session,
 
 
 
-            if (url [strlen (url) -1] != '/') strncat (filepath, "/", sizeof (filepath));
-            strncat (filepath, "index.html", sizeof (filepath));
+            if (url [strlen (url) -1] != '/') strncat (filepath, "/", sizeof (filepath) - 1);
+            strncat (filepath, "index.html", sizeof (filepath) - 1);
             close (fd);
             response = MHD_create_response_from_buffer (0,"", MHD_RESPMEM_PERSISTENT);
             MHD_add_response_header (response,MHD_HTTP_HEADER_LOCATION, filepath);

--- a/src/session-ajq.c
+++ b/src/session-ajq.c
@@ -1,7 +1,7 @@
 /*
    alsajson-gw -- provide a REST/HTTP interface to ALSA-Mixer
 
-   Copyright (C) 2015, Fulup Ar Foll
+   Copyright (C) 2018, Fulup Ar Foll, Google LLC
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -162,12 +162,12 @@ STATIC void makeSessionLink (const char *cardname, const char *sessionname) {
    int err;
    // create a link to keep track of last uploaded sessionname for this card
    strncpy (filename, sessionname, sizeof(filename));
-   strncat (filename, ".ajg", sizeof(filename));
+   strncat (filename, ".ajg", sizeof(filename) - 1);
 
    strncpy (linkname, cardname, sizeof(linkname));
-   strncat (linkname, "/", sizeof(filename));
-   strncat (linkname, AJG_CURRENT_SESSION, sizeof(linkname));
-   strncat (linkname, ".ajg", sizeof(filename));
+   strncat (linkname, "/", sizeof(filename) - 1);
+   strncat (linkname, AJG_CURRENT_SESSION, sizeof(linkname) - 1);
+   strncat (linkname, ".ajg", sizeof(filename) - 1);
    unlink (linkname); // remove previous link if any
    err = symlink (filename, linkname);
    if (err < 0) fprintf (stderr, "Fail to create link %s->%s error=%s\n", linkname, filename, strerror(errno));
@@ -193,10 +193,10 @@ PUBLIC json_object *sessionFromDisk (AJG_session *session, AJG_request *request)
 
     // add cardname and file extension to session name
     strncpy (filename, request->cardname, sizeof(filename));
-    strncat (filename, "/", sizeof(filename));
+    strncat (filename, "/", sizeof(filename) - 1);
     if (defsession) strncat (filename, AJG_CURRENT_SESSION, sizeof(filename)-1);
     else strncat (filename, request->args, sizeof(filename)-1);
-    strncat (filename, ".ajg", sizeof(filename));
+    strncat (filename, ".ajg", sizeof(filename) - 1);
 
     // just upload json object and return without any further processing
     jsonSession = json_object_from_file (filename);
@@ -242,7 +242,7 @@ PUBLIC json_object * sessionToDisk (AJG_session *session, AJG_request *request, 
 
    // add cardname and file extension to session name
    strncpy (filename, request->cardname, sizeof(filename));
-   strncat (filename, "/", sizeof(filename));
+   strncat (filename, "/", sizeof(filename) - 1);
    if (defsession) strncat (filename, AJG_CURRENT_SESSION, sizeof(filename)-1);
    else strncat (filename, request->args, sizeof(filename)-1);
    strncat (filename, ".ajg", sizeof(filename)-1);

